coverage/PENDING_SUBMIT/swiftshader/0338-COVERAGE-apfloat-2108-apint-2371/cov-undefined-clamp.amber

```diff
@@ -0,0 +1,230 @@
+#!amber
+
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# A test for a coverage-gap found by the GraphicsFuzz project.
+
+# Short description: A fragment shader that covers specific APFloat and APInt code paths
+
+# The test passes because the shader always writes red.
+
+SHADER vertex variant_vertex_shader PASSTHROUGH
+
+# variant_fragment_shader is derived from the following GLSL:
+# #version 310 es
+# #define _int_1 _GLF_uniform_int_values[0]
+# #define _int_0 _GLF_uniform_int_values[1]
+# #define _int_2 _GLF_uniform_int_values[2]
+# #define _float_0_2 _GLF_uniform_float_values[0]
+# #define _float_1_0 _GLF_uniform_float_values[1]
+# #define _float_0_0 _GLF_uniform_float_values[2]
+#
+# precision highp int;
+# precision highp float;
+#
+# // Contents of _GLF_uniform_float_values: [0.2, 1.0, 0.0]
+# layout(set = 0, binding = 0) uniform buf0
+# {
+#     float _GLF_uniform_float_values[3];
+# };
+#
+# // Contents of _GLF_uniform_int_values: [1, 0, 2]
+# layout(set = 0, binding = 1) uniform buf1
+# {
+#     int _GLF_uniform_int_values[3];
+# };
+#
+# layout(location = 0) out vec4 _GLF_color;
+#
+# void main()
+# {
+#     vec2 c;
+#
+#     // Z component is undefined but doesn't matter because only the x and y components
+#     // are used for setting the color.
+#     vec3 v1 = vec3(_float_1_0, _float_0_0, int(c.x - 1.0));
```

Comment:
My concern is that doing `c.x - 1.0` does mean that the uninitialized variable is accessed. If a compiler treats this using "catch fire" semantics, it might assume that the access cannot be reached, effectively allowing the compiler to optimize away "main".

Reply:
I think we have to abandon this test. There seems to be no way to get rid of the access of `c.x` or initialize it without the test becoming uninteresting.

Reply:
I feel like we previously decided that uninitialized values remain nondeterministic forever, as opposed to "catch fire" semantics. But, either way makes sense to me. Have you tried adding something like this before `vec3 v1 = ...;`:

```
if (_int_0 == 0) {
  c.x = 0.0;
}
```

Reply:
> ```
> if (_int_0 == 0) {
>   c.x = 0.0;
> }
> ```

Unfortunately, this also makes the COV message disappear.